lib/backend/k8s/conversion.go

```diff
@@ -167,6 +167,35 @@ func (c Converter) PodToWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
 	}
 	labels["calico/k8s_ns"] = pod.ObjectMeta.Namespace
 
+	// Map any named ports through.
+	var endpointPorts []model.EndpointPort
+	for _, container := range pod.Spec.Containers {
+		for _, containerPort := range container.Ports {
```

Comment:
We seem to be losing that - in the k8s data structures - a named port belongs to a *particular* container within the pod.  Does that matter?

Reply:
I don't think so, the containers share a network namespace and k8s enforces uniqueness across a whole pod.  I think the association with containers in the k8s model is fairly natural because ports belong to the particular process and hence the particular container but, since we don't have the concept of a sub-workload in our datamodel, I don't think we lose anything.